### PR TITLE
Tweaks for batch update

### PR DESCRIPTION
**Pull request by bitwiseman**
# Description 
** Describe your change here** 

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [ ] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [ ] Add JavaDocs and other comments
- [ ] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [ ] Run `mvn clean compile` locally. This may reformat your code, commit those changes.
- [ ] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHDiscussion.java [hunk 1]
```diff
@@ -63,68 +69,141 @@ public String getBody() {
      *
      * @return the number
      */
-    public String getNumber() {
+    public long getNumber() {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHDiscussion.java:74]: This should probably be replaced by `GHObject.getId()` somehow.

Comment (cmoulliard) [reply src/main/java/org/kohsuke/github/GHDiscussion.java:74]: Not sure as the API uses as field `number` and not at all `id`. See example reported here as response: https://developer.github.com/v3/teams/discussions/#get-a-discussion

File: src/main/java/org/kohsuke/github/GHDiscussion.java [hunk 2]
```diff
@@ -63,68 +69,141 @@ public String getBody() {
...
      *             the io exception
      */
-    public void delete(String number) throws IOException {
-        root.createRequest()
-                // .withPreview(INERTIA)
+    public void delete() throws IOException {
+        team.root.createRequest()
                 .method("DELETE")
-                .withUrlPath("/orgs/" + team.getOrganization().getLogin() + "/teams/" + team.getSlug() + "/discussions/"
-                        + number)
```
Comment (cmoulliard) [inline src/main/java/org/kohsuke/github/GHDiscussion.java:127]: We should perhaps have a method able to populate the part `/orgs/org/teams/team_slug/discussions` OR `/organizations/org_id/teams/team_id/discussions` which is used by all the CRUD operations. WDYT ?

File: src/test/java/org/kohsuke/github/junit/GitHubWireMockRule.java [hunk 3]
```diff
@@ -110,43 +110,33 @@ protected void after() {
             return;
         }
 
-        // "If-None-Match" header used for ETag matching for caching connections
-        // "Accept" header is used to specify previews. If it changes expected data may not be retrieved.
-        this.apiServer()
```
Comment (cmoulliard) [inline src/test/java/org/kohsuke/github/junit/GitHubWireMockRule.java]: Why do you changed the class `GitHubWireMockRule.java` part of this Discussion PR ?

File: src/main/java/org/kohsuke/github/GHDiscussion.java (hunk 1 above)
Comment (bitwiseman) [reply src/main/java/org/kohsuke/github/GHDiscussion.java:74]: It is effectively the same as `id`.  I'll add an override of `getId()` that calls `getNumber()`.

File: src/main/java/org/kohsuke/github/GHDiscussion.java (hunk 2 above)
Comment (bitwiseman) [reply src/main/java/org/kohsuke/github/GHDiscussion.java:127]: Sure, like other classes have.

File: src/test/java/org/kohsuke/github/junit/GitHubWireMockRule.java (hunk 3 above)
Comment (bitwiseman) [reply src/test/java/org/kohsuke/github/junit/GitHubWireMockRule.java]: This change is needed to make it so the playback in GHDiscussionTest works correctly.  You are correct.  I've committed separately now and merged.

File: src/main/java/org/kohsuke/github/GHDiscussion.java (hunk 2 above)
Comment (bitwiseman) [reply src/main/java/org/kohsuke/github/GHDiscussion.java:127]: Done.

**Comment by bitwiseman** [comment]
@cmoulliard Okay, you can merge this if it looks good to you.

**Pull request merged**